spring-xd-dirt/src/main/java/org/springframework/xd/dirt/module/WriteCapableArchiveModuleRegistry.java

```diff
@@ -42,13 +42,13 @@ public WriteCapableArchiveModuleRegistry(String root) {
 	@Override
 	public boolean delete(ModuleDefinition definition) {
 		try {
-			File module = targetModuleLocation(definition);
-			boolean wasFile = module.exists() && !module.isDirectory() && module.delete();
-			if (wasFile) {
-				return true;
-			} else {
-				String filename = module.getName();
-				File asDir = new File(module.getParentFile(), filename.substring(0, filename.lastIndexOf('.')));
+			File moduleJarFile = targetModuleLocation(definition);
+			if (moduleJarFile.exists()) {
```

Comment:
I kinda take that back. targetModuleLocation() would return "xxx.jar" path, so it would not exist...

... UNLESS the module directory itself was named xxx.jar (which is currently supported, but would like to remove)